soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4055,27 +4069,42 @@ void DrawRandoEditor(bool& open) {
 
                     // Cuccos to return
                     SohImGui::EnhancementSliderInt("Cuccos to return: %d", "##RandoCuccosToReturn",
-                                                    "gRandomizeCuccosToReturn", 0, 7, "", 7);
+                                                    "gRandomizeCuccosToReturn", 0, 7, "", 7, true);
                     InsertHelpHoverText(
                         "The amount of cuccos needed to claim the reward from Anju the cucco lady"
                     );
                     ImGui::Separator();
 
                     // Big Poe Target Count
                     SohImGui::EnhancementSliderInt("Big Poe Target Count: %d", "##RandoBigPoeTargetCount",
-                                                    "gRandomizeBigPoeTargetCount", 1, 10, "", 10);
+                                                    "gRandomizeBigPoeTargetCount", 1, 10, "", 10, true);
                     InsertHelpHoverText(
                         "The Poe collector will give a reward for turning in this many Big Poes."
                     );
                     ImGui::Separator();
 
                     // Skip child stealth
-                    SohImGui::EnhancementCheckbox(Settings::SkipChildStealth.GetName().c_str(), "gRandomizeSkipChildStealth");
-                    InsertHelpHoverText(
-                        "The crawlspace into Hyrule Castle goes straight to Zelda, skipping\n"
-                        "the guards."
-                    );
+                    // Disabled when Skip Child Zelda is active
+                    if (disableZeldaRelatedOptions) {
+                        CVar_SetS32("gRandomizeSkipChildStealth", 0);
+                        // Also disable Weird Egg because it's on a different tab and wouldn't be updated otherwise
+                        CVar_SetS32("gRandomizeShuffleWeirdEgg", 0);
+                    }
+                    ImGui::PushItemFlag(ImGuiItemFlags_Disabled, disableZeldaRelatedOptions);
+                    ImGui::PushStyleVar(ImGuiStyleVar_Alpha,
+                                        ImGui::GetStyle().Alpha * disableZeldaRelatedOptions ? 0.5f : 1.0f);
+                    SohImGui::EnhancementCheckbox(Settings::SkipChildStealth.GetName().c_str(),
```

Comment:
is the plan to use the names from 3drando settings like this? if so, more of these should be updated. if not (especially in the case of skip child zelda where we don't have a 3drando setting for it) we should not use it and should instead look into a better solution

if the plan is to try to bypass the 3drando settings class at some point, then adding more things that rely on it will make that more difficult

Reply:
I changed this before we talked about not using the Option class at all and I was still in the mindset that we'd change the Option class around instead. Most settings do it this way currently after my first UX PR, that's why I had to change some of the names in the settings.cpp part. I could manually replace all of these with what's currently in the Option class though if it would make it easier to transition away from the Option class in the future. 

Lemme know.

Reply:
If you want to start pulling those strings out into macros or something and pull them into 3d rando settings and imgui that could be cool

Reply:
I think that would be better suited for a seperate PR.

Reply:
that works